src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -271,33 +274,35 @@ private static void CheckDefaultProvider(Type type)
             {
                 if (s_defaultProviders.ContainsKey(type))
                 {
+                    // Either another thread finished initializing for this type, or we are recursing on the same thread.
                     return;
                 }
 
-                // Immediately clear this. If we find a default provider
-                // and it starts messing around with type information,
+                // Immediately set this so we can short circuit in the ContainsKey check above.
+                // If we find a default provider and it starts messing around with type information,
```

Comment:
I still don't understand this comment.  This is how I would word it.
```suggestion
                // Immediately set this to null to indicate we are in progress setting the default provider for a type.
                // This prevents re-entrance in this method.
```

Reply:
Thanks - added